src/bot.rs

```diff
@@ -331,13 +333,97 @@ impl RcBot {
                     sender
                         .unbounded_send((self.clone(), val.message.unwrap()))
                         .unwrap_or_else(|e| error!("Error: {}", e));
-                    return None;
+                    None
+                } else if forward {
+                    Some((self.clone(), val))
                 } else {
-                    return Some((self.clone(), val));
+                    None
                 }
             })
     }
 
+    /// Similar to `get_push_stream`, but takes a function that takes an
+    /// update and returns a future that returns a response.
+    ///
+    /// It is possible to return a request directly in the response.
+    pub fn get_push_stream_ex<'a, F, G>(&'a self, f: &'static F)
+        -> impl Stream<Item = (RcBot, objects::Update), Error = Error> + 'a
+        where
+            F: Send + Sync + Fn(&objects::Update) -> (G, bool),
+            G: 'static + IntoFuture<Item = Response<Body>, Error = Error>,
```

Comment:
The fact that this function has to be passed into Hyper means it has to be `'static`, which kind of makes using a closure here really difficult. Also it's impossible to pass an `RcBot` into it, because Hyper seems to involve threads (and so we need an `Arc` at least? Not sure.)

Not really happy with this but I'm not sure what else to do either.

Reply:
As I wrote earlier the the serve function should be single threaded, but needs an executor (like in https://github.com/hyperium/hyper/blob/master/examples/single_threaded.rs)  For what reason do we need different push handlers, can't we just assume that the reply is always the same? 
The code structure is getting really complicated lately, but I see no easier way :smile: